python/rpdk/java/templates/init/guided_aws/TagHelper.java

```diff
@@ -0,0 +1,256 @@
+package {{ package_name }};
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.google.common.collect.Sets;
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.MapUtils;
+import org.apache.commons.lang3.ObjectUtils;
+import software.amazon.awssdk.awscore.AwsResponse;
+import software.amazon.awssdk.core.SdkClient;
+// TODO: Replace the Tag model below according to your resource
+import software.amazon.awssdk.services.cloudformation.model.Tag;
+import software.amazon.cloudformation.proxy.AmazonWebServicesClientProxy;
+import software.amazon.cloudformation.proxy.Logger;
+import software.amazon.cloudformation.proxy.ProgressEvent;
+import software.amazon.cloudformation.proxy.ProxyClient;
+import software.amazon.cloudformation.proxy.ResourceHandlerRequest;
+
+public class TagHelper {
+    /**
+     * convertToMap
+     *
+     * Converts a collection of Tag objects to a tag-name -> tag-value map.
+     *
+     * Note: Tag objects with null tag values will not be included in the output
+     * map.
+     *
+     * @param tags Collection of tags to convert
+     * @return Converted Map of tags
+     */
+    public static Map<String, String> convertToMap(final Collection<Tag> tags) {
+        if (CollectionUtils.isEmpty(tags)) {
+            return Collections.emptyMap();
+        }
+        return tags.stream()
+            .filter(tag -> tag.value() != null)
+            .collect(Collectors.toMap(
+                Tag::key,
+                Tag::value,
+                (oldValue, newValue) -> newValue));
+    }
+
+    /**
+     * convertToSet
+     *
+     * Converts a tag map to a set of Tag objects.
+     *
+     * Note: Like convertToMap, convertToSet filters out value-less tag entries.
+     *
+     * @param tagMap Map of tags to convert
+     * @return Set of Tag objects
+     */
+    public static Set<Tag> convertToSet(final Map<String, String> tagMap) {
+        if (MapUtils.isEmpty(tagMap)) {
+            return Collections.emptySet();
+        }
+        return tagMap.entrySet().stream()
+            .filter(tag -> tag.getValue() != null)
+            .map(tag -> Tag.builder()
+                .key(tag.getKey())
+                .value(tag.getValue())
+                .build())
+            .collect(Collectors.toSet());
+    }
+
+    /**
+     * generateTagsForCreate
+     *
+     * Generate tags to put into resource creation request.
+     * This includes user defined tags and system tags as well.
+     */
+    public final Map<String, String> generateTagsForCreate(final ResourceModel resourceModel, final ResourceHandlerRequest<ResourceModel> handlerRequest) {
+        final Map<String, String> tagMap = new HashMap<>();
+
+        if (handlerRequest.getSystemTags() != null) {
```

Comment:
Generally, the `systemTags` won't be null in most of the cases.  Whether to put into the desired tags is based on whether the service support system tags.  I think this will cause more problems. May just need to comment it out and have a comment on it.

Reply:
sure, added this comment